app/src/main/java/com/jetpack/compose/learning/maps/geojson/MapGeoJsonActivity.kt

```diff
@@ -0,0 +1,127 @@
+import com.jetpack.compose.learning.maps.currentMarkerLatLong
+import com.jetpack.compose.learning.maps.getFeatureBound
+import com.jetpack.compose.learning.maps.setFeatureStyle
+import com.jetpack.compose.learning.theme.AppThemeState
+import com.jetpack.compose.learning.theme.BaseView
+import com.jetpack.compose.learning.theme.SystemUiController
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+/**
+ * GeoJson Example.
+ * It includes the MapEffect side effect to get the map object and create geo json layer.
+ */
+class MapGeoJsonActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            val systemUiController = remember { SystemUiController(window) }
+            val appTheme = remember { mutableStateOf(AppThemeState()) }
+            BaseView(appTheme.value, systemUiController) {
+                MainContent()
+            }
+        }
+    }
+
+    @Composable
+    fun MainContent() {
+        var showLoading by remember { mutableStateOf(true) }
+
+        MapScaffold(
+            title = "GeoJSON",
+            onBackPressed = { onBackPressed() },
+            showLoading = showLoading
+        ) {
+            Text(
+                "Long click on map to add your custom feature point.",
+                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
+                style = MaterialTheme.typography.subtitle1,
+                textAlign = TextAlign.Center
+            )
+            MapsExample { showLoading = false }
+        }
+    }
+
+    @OptIn(MapsComposeExperimentalApi::class)
+    @Composable
+    fun MapsExample(onMapLoaded: () -> Unit) {
+        val cameraPositionState = rememberCameraPositionState {
+            position = CameraPosition.fromLatLngZoom(currentMarkerLatLong, 17f)
+        }
+        var geoJsonLayer by remember { mutableStateOf<GeoJsonLayer?>(null) }
+        val context = LocalContext.current
+        val scope = rememberCoroutineScope()
+
+        GoogleMap(
+            modifier = Modifier.fillMaxSize(),
+            cameraPositionState,
+            onMapLoaded = onMapLoaded,
+            onMapLongClick = {
+                if (geoJsonLayer != null) {
+                    val point = GeoJsonPoint(it)
+                    val properties = hashMapOf("title" to "My Custom Location")
+                    val pointFeature = GeoJsonFeature(point, "CU-$it", properties, null)
+                    val pointStyle = GeoJsonPointStyle()
+                    pointStyle.title = properties["title"]
+                    pointFeature.pointStyle = pointStyle
+                    geoJsonLayer?.addFeature(pointFeature)
+                }
+            }
+        ) {
+            MapEffect(Unit) { map ->
+                if (geoJsonLayer == null) {
+                    geoJsonLayer = GeoJsonLayer(map, R.raw.geo_json_sample, context)
```

Comment:
![Screenshot 2022-07-27 at 6 39 52 PM](https://user-images.githubusercontent.com/66367742/181255476-bc8486c9-326e-4720-b9ba-3f37328b36da.png)

Can we check this warning?